### DISCUSSION TITLE
Use TalkBack for TTS events on Android

### DISCUSSION THREAD

**Discussion by CoBC** [BearWare/TeamTalk5#1413] [Ideas]
Hi,
currently, Android only support TTS via the TTS API. A method (announceForAccessibility, introduced in API level 16) can allow usage of accessibility service like TalkBack to announce TTS events.
Using TalkBack to announce TTS events can allow users to easily interrupt the notification if necessary. I propose to add an option to allow users to use TalkBack instead of native TTS API, so the behavior will be the same as using Tolk on Windows for example.
What Android users think of that? @bear101 Do you also have any point of view of that?

**Comment by Nardol** [comment]
It would be a very good enhancement.
Having announcements separate from TalkBack make them sometimes interfere between them.

**Comment by lbk2907** [comment]
Not all TeamTalk user using screen reader.
Is better to add the option for it.
Suggestion: Add an ability for TeamTalk to use an accessibility volume for
the text-To-Speech event.
Bug Report: For now, if user want to choose TTS that they want to use to
read Text-To-Speech event, they need to connect to server at least once
each time they open the TeamTalk. If user directly open the TTS selection
without connect to any server at least once, the list will be empty. They
will just have the cancel button there.

**Comment by beqabeqa473** [comment]
using tb for announcing notifications might be an option but not a
replacement, because it won't announce anything when screen is off.

On 7/31/22, lbk2907 ***@***.***> wrote:
> Not all TeamTalk user using screen reader.
> Is better to add the option for it.
> Suggestion: Add an ability for TeamTalk to use an accessibility volume for
> the text-To-Speech event.
> Bug Report: For now, if user want to choose TTS that they want to use to
> read Text-To-Speech event, they need to connect to server at least once
> each time they open the TeamTalk. If user directly open the TTS selection
> without connect to any server at least once, the list will be empty. They
> will just have the cancel button there.
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/1413#discussioncomment-3290776
> You are receiving this because you are subscribed to this thread.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by lbk2907** [comment]
I agree with this.

**Comment by bear101** [comment]
@CoBC It would be nice to also have the option to choose full message or interrupt. You're welcome to make a Pull-Request with this new feature :)

**Comment by lbk2907** [comment]
@CoBC

Are possible to add the speech rate also?

User have an option to choose whether they want to follow the system
settings or they can customize it.

Same with the TTS engine selection, they have that option too.

Thanks.

**Comment by alessio0292** [comment]
hey my new friend @CoBC do something for me because in this moment Bjoern banned me in github and I can't speak with Bjoern and other, nardol banned me here in a discussion and the discussion i repeat, I'd like a different team talk and for this thing i suggested many thing and i'd like to do many other requests!

**Comment by alessio0292** [comment]
hey my new friend @CoBC do something for me because in this moment Bjoern banned me in github and I can't speak with Bjoern and other, nardol banned me here in a discussion and the discussions i repeat, I'd like a different team talk and for this thing i suggested many things and i'd like to do many other requests!
